src/main/java/org/mastodon/mamut/feature/branch/divisions/BranchCellDivisionFrequencyFeatureComputer.java

```diff
@@ -28,65 +28,88 @@
  */
 package org.mastodon.mamut.feature.branch.divisions;
 
-import org.mastodon.mamut.feature.AbstractSerialFeatureComputer;
+import net.imglib2.util.Pair;
+import net.imglib2.util.ValuePair;
+import org.mastodon.mamut.feature.AbstractResettableFeatureComputer;
 import org.mastodon.mamut.feature.MamutFeatureComputer;
-import org.mastodon.mamut.feature.ValueIsSetEvaluator;
 import org.mastodon.mamut.feature.branch.BranchSpotFeatureUtils;
+import org.mastodon.mamut.model.branch.BranchLink;
 import org.mastodon.mamut.model.branch.BranchSpot;
+import org.mastodon.mamut.model.branch.ModelBranchGraph;
+import org.mastodon.mamut.util.LineageTreeUtils;
 import org.mastodon.properties.DoublePropertyMap;
+import org.mastodon.properties.ObjPropertyMap;
 import org.scijava.ItemIO;
 import org.scijava.plugin.Parameter;
 import org.scijava.plugin.Plugin;
 
-import java.util.Collection;
+import javax.annotation.Nonnull;
 
 /**
  * Computes {@link BranchCellDivisionFrequencyFeature}
  */
 @Plugin( type = MamutFeatureComputer.class )
-public class BranchCellDivisionFrequencyFeatureComputer extends AbstractSerialFeatureComputer< BranchSpot >
+public class BranchCellDivisionFrequencyFeatureComputer extends AbstractResettableFeatureComputer
 {
 
+	@Parameter
+	protected ModelBranchGraph branchGraph;
+
 	@Parameter( type = ItemIO.OUTPUT )
 	protected BranchCellDivisionFrequencyFeature output;
 
-	@Override
-	protected void compute( final BranchSpot branchSpot )
-	{
-		output.nCellDivisions.set( branchSpot, cellDivisionFrequency( branchSpot ) );
-	}
+	private ObjPropertyMap< BranchSpot, Pair< Integer, Integer > > valueCache;
```

Comment:
I would use a `RefObjectMap` instead of the `ObjPropertyMap` but I barely understand the difference between the two classes.